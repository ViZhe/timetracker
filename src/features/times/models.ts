
export interface ITimesData {
  description: string;
  duration: string;
  key: string;
  tags: string[];
  timeStart: string;
  timeEnd: string | null;
}

export interface ITimesState {
  data: ITimesData[];
  isLoading: boolean;
}
