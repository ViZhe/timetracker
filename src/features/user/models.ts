
export interface IUserState {
  name: string | null;
}
