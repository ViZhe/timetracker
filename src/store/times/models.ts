
export interface ITimesData {
  description: string;
  duration: string;
  key: string;
  tags: string[];
  timeStart: Date | string;
  timeEnd: Date | string;
}

export interface ITimesState {
  data: ITimesData[];
  isLoading: boolean;
}
