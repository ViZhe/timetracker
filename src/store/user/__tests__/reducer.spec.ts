
import {
  reducer,
  userActions as actions,
} from '../';
import { IUserState } from '../models';


describe('User Reducer', () => {
  const getInitialState = (initialState?: Partial<IUserState>) => (
    reducer(initialState as IUserState, {} as any)
  );

  describe('User Reducer initial', () => {
    const initialState = getInitialState();

    it('should match a snapshot', () => {
      expect(initialState).toMatchSnapshot();
    });

    it('should set a user\'s data', () => {
      const payload = { name: 'test' };

      expect(initialState.name).toBeNull();
      const newState = reducer(initialState, actions.setData(payload));
      expect(newState.name).toEqual(payload.name);
    });
  });
});
