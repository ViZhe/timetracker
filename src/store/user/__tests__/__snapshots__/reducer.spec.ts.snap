// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Reducer User Reducer initial should match a snapshot 1`] = `
Object {
  "name": null,
}
`;
